### PR TITLE
Remove deprecation for pulse instruction parameter method

### PR DESCRIPTION
**Pull request by nkanazawa1989**
### Summary
The parameter manager (in PR #5854) deprecated the instruction-level management of parameters to realize light-weight instruction object, i.e.

```python
inst = Play(Gaussian(**params), d0)
inst.parameters  # deprecated
```

However this method is still needed for writing calibration code (thanks @eggerdj !). This PR removes deprecation warnings and add new logic to get the parameter set without initializing parameter table in the constructor.

### Details and comments
No unittest is added by this PR, because there are tests for `.parameters` method in `test/python/pulse/test_parameters.py`. These tests were not removed by #5854 for backward compatibility thus they should be still valid.

### CODE REVIEW

**Comment by mtreinish** [comment]
Can you add a release note here? Probably an `other` note saying this isn't deprecated anymore. Since IIRC there was release note about this deprecation, we should document we're reversing that and it's not deprecated anymore and will continue to be supported.

**Comment by nkanazawa1989** [comment]
Thanks @mtreinish for catching this. Yes, deprecation node was added to 0.17 release. Reno is added.

**Review by eggerdj — APPROVED** [review]
LGTM.

File: qiskit/pulse/instructions/play.py [hunk 1]
```diff
@@ -93,6 +93,20 @@ def _initialize_parameter_table(self,
...
+    @property
+    def parameters(self) -> Set:
+        """Parameters which determine the instruction behavior."""
+        parameters = set()
+        for pulse_param_expr in self.pulse.parameters.values():
+            if isinstance(pulse_param_expr, ParameterExpression):
+                for pulse_param in pulse_param_expr.parameters:
+                    parameters.add(pulse_param)
+        if self.channel.is_parameterized():
+            for ch_param in self.channel.parameters:
```
Comment (1ucian0) [inline qiskit/pulse/instructions/play.py:105]: It seems that the tests need to be extended with parametrised channels:  https://coveralls.io/builds/39061954/source?filename=qiskit%2Fpulse%2Finstructions%2Fplay.py#L105

**Review by 1ucian0 — CHANGES_REQUESTED** [review]
Not fully sure, but maybe a new test is needed to cover the added code.

**Comment by nkanazawa1989** [comment]
Thanks @1ucian0 for catching this. Though these functionalities have been existing, it seems like we didn't have specific test for them. I added new test for play instruction paramters.

**Review by 1ucian0 — APPROVED** [review]
Thanks!

**Pull request merged**